src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponent.scala

```diff
@@ -26,6 +26,15 @@ trait DriverComponent {
     }
   }
 
+  def uniqueResult[V](results: ReadAction[Seq[V]]): ReadAction[Option[V]] = {
+    results map {
+      case Seq() => None
+      case Seq(one) => Option(one)
+      case tooMany => throw new RawlsException(s"Expected 0 or 1 result but found all of these: $tooMany")
+    }
+  }
+
```

Comment:
a little code duplication but could not figure out how to eliminate it (quickly anyway)